src/dds/rs-dds-sensor-proxy.h

```diff
@@ -115,6 +115,9 @@ class dds_sensor_proxy : public software_sensor
     virtual void add_no_metadata( frame *, streaming_impl & );
     virtual void add_frame_metadata( frame *, rsutils::json const & metadata, streaming_impl & );
 
+    void add_processing_block_settings( const std::string & filter_name,
+                                        std::shared_ptr< librealsense::processing_block_interface > & ppb ) const;
```

Comment:
why do you need here shared_ptr by reference? It seems to me that you want it by value, so that the copy ctor works, no?

Reply:
No, no need to create a new shared_ptr and update the reference count.
It's just a helper function that uses the shared_ptr of the caller, we don't want to store it.

Reply:
ok